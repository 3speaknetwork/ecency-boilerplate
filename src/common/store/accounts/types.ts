import { ExtendedAccount } from "@esteemapp/dhive";

export interface Account extends ExtendedAccount {}

export interface State {
  data: Account;
  loading: boolean;
  error: boolean;
}

export enum ActionTypes {
  FETCH = "@account/FETCH",
  FETCHED = "@account/FETCHED",
  FETCH_ERROR = "@account/FETCH_ERROR"
}

export interface FetchAction {
  type: ActionTypes.FETCH;
}

export interface FetchedAction {
  type: ActionTypes.FETCHED;
  data: Account;
}

export interface FetchErrorAction {
  type: ActionTypes.FETCH_ERROR;
}

export type Actions = FetchAction | FetchedAction | FetchErrorAction;