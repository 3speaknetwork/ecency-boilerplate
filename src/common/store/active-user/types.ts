import { Account } from "../accounts/types";

export interface ActiveUser {
  name: string;
  data: Account;
}

export enum ActionTypes {
  RELOAD = "@active-user/RELOAD",
  UPDATE = "@active-user/UPDATE",
}

export interface ReloadAction {
  type: ActionTypes.RELOAD;
}

export interface UpdateAction {
  type: ActionTypes.UPDATE;
  data: Account;
}

export type Actions = ReloadAction | UpdateAction;
