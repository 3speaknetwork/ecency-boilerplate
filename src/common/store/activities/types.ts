interface BaseActivity {
    event: "notify";
    source: string;
    target: string;
    timestamp: string;
    extra: any;
}

export interface VoteActivity extends BaseActivity {
    type: "vote",
    extra: {
        permlink: string;
        weight: number;
        title: string | null;
        img_url: string | null;
    }
}

export interface MentionActivity extends BaseActivity {
    type: "mention",
    extra: {
        permlink: string;
        is_post: 0 | 1;
        title: string | null;
        img_url: string | null;
    }
}

export interface FollowActivity extends BaseActivity {
    type: "follow",
    extra: {
        what: string[]
    }
}

export interface ReplyActivity extends BaseActivity {
    type: "reply",
    extra: {
        title: string;
        body: string;
        json_metadata: string;
        permlink: string;
        parent_author: string;
        parent_permlink: string;
        parent_title: string | null;
        parent_img_url: string | null;
    }
}

export interface ReblogActivity extends BaseActivity {
    type: "reblog";
    extra: {
        permlink: string;
        title: string | null;
        img_url: string | null;
    }

}

export interface TransferActivity extends BaseActivity {
    type: "transfer";
    extra: {
        amount: string;
        memo: string;
    }
}


export type Activity = VoteActivity | MentionActivity | FollowActivity | ReplyActivity | ReblogActivity | TransferActivity;
