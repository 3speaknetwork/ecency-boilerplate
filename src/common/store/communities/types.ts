export interface Community {
    about: string,
    avatar_url: string,
    created_at: string,
    description: string,
    flag_text: string,
    id: number,
    is_nsfw: boolean,
    lang: string,
    name: string,
    num_authors: number,
    num_pending: number,
    subscribers: number,
    sum_pending: number,
    team: Array<Array<string>>,
    title: string,
    type_id: number
}

export interface State {
    list: Record<string, Community>,
}

export enum ActionTypes {
    FETCHED = '@communities/FETCHED',
}


export interface FetchedAction {
    type: ActionTypes.FETCHED;
    name: string,
    data: Community
}

export type Actions = FetchedAction; // | .. | ..
