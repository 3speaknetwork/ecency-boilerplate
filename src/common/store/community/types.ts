export interface Community {
  about: string;
  avatar_url: string;
  created_at: string;
  description: string;
  flag_text: string;
  id: number;
  is_nsfw: boolean;
  lang: string;
  name: string;
  num_authors: number;
  num_pending: number;
  subscribers: number;
  sum_pending: number;
  settings?: any;
  team: Array<Array<string>>;
  title: string;
  type_id: number;
}

export interface State extends Community {}

export enum ActionTypes {
  FETCHED = "@communities/FETCHED",
  RESET = "@communities/RESET",
}

export interface FetchedAction {
  type: ActionTypes.FETCHED;
  data: Community;
}

export interface ResetAction {
  type: ActionTypes.RESET;
}

export type Actions = FetchedAction | ResetAction;
