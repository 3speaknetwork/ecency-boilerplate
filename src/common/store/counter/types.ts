export interface State {
    val: number
}

export enum ActionTypes {
    SET_COUNTER = '@counter/SET_COUNTER',
    INCREMENT_COUNTER = '@counter/INCREMENT_COUNTER',
    DECREMENT_COUNTER = '@counter/DECREMENT_COUNTER',
}

export interface SetAction {
    type: ActionTypes.SET_COUNTER;
    value: string;
}

export interface IncrementAction {
    type: ActionTypes.INCREMENT_COUNTER;
}

export interface DecrementAction {
    type: ActionTypes.DECREMENT_COUNTER;
}

export type Actions = SetAction | IncrementAction | DecrementAction
