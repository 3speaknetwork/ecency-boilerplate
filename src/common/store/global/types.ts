import { LocationChangeAction } from "../common";

export enum ListStyle {
  row = "row",
  grid = "grid",
}

export enum Theme {
  day = "day",
  night = "night",
}

export enum EntryFilter {
  trending = "trending",
  hot = "hot",
  created = "created",
  payout = "payout",
}

export enum ProfileFilter {
  posts = "posts",
  comments = "comments",
  replies = "replies",
}

// TODO: Find a proper way to merge EntryFilter and ProfileFilter
export enum AllFilter {
  trending = "trending",
  hot = "hot",
  created = "created",
  payout = "payout",
  muted = "muted",  // To see muted accounts
  blog = "blog",  // This might be deleted
  posts = "posts",
  comments = "comments",
  replies = "replies",
  feed = "feed"
}

export interface Global {
  filter: EntryFilter | ProfileFilter | AllFilter;
  tag: string;
  theme: Theme;
  listStyle: ListStyle;
  intro: boolean;
  currency: string;
  currencyRate: number;
  currencySymbol: string;
  searchIndexCount: number;
  canUseWebp: boolean;
}

export enum ActionTypes {
  THEME_CHANGE = "@global/THEME_CHANGE",
  INTRO_HIDE = "@global/INTRO_HIDE",
  LIST_STYLE_CHANGE = "@global/LIST_STYLE_CHANGE",
}

export interface ThemeChangeAction {
  type: ActionTypes.THEME_CHANGE;
  theme: Theme;
}

export interface IntroHideAction {
  type: ActionTypes.INTRO_HIDE;
}

export interface ListStyleChangeAction {
  type: ActionTypes.LIST_STYLE_CHANGE;
  listStyle: ListStyle;
}

export type Actions = LocationChangeAction | ThemeChangeAction | IntroHideAction | ListStyleChangeAction;
