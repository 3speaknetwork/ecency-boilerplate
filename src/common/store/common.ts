export enum CommonActionTypes {
  INIT = "@@INIT",
  LOCATION_CHANGE = "@@router/LOCATION_CHANGE",
}

export interface LocationChangeAction {
  type: CommonActionTypes.LOCATION_CHANGE;
  payload: {
    location: {
      pathname: string;
    };
  };
}

export interface InitAction {
  type: CommonActionTypes.INIT;
}

export const locationChangeAct = (pathname: string): LocationChangeAction => {
  return {
    type: CommonActionTypes.LOCATION_CHANGE,
    payload: {
      location: {
        pathname,
      },
    },
  };
};
