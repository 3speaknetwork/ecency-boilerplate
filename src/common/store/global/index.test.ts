import reducer, {initialState, themeChangeAct, hideIntroAct, listStyleChangeAct} from './index';

import {State, Theme, ListStyle} from './types';
import {locationChangeAct} from '../common';

let state: State = initialState;

it('1- default state', () => {
    expect(state).toMatchSnapshot();
});

it('2- location change. should change filter and tag', () => {
    state = reducer(state, locationChangeAct('/trending/bitcoin'));
    expect(state).toMatchSnapshot();
});

it('3- theme change', () => {
    state = reducer(state, themeChangeAct(Theme.night));
    expect(state).toMatchSnapshot();
});

it('4- hide intro', () => {
    state = reducer(state, hideIntroAct());
    expect(state).toMatchSnapshot();
});

it('5- list style change', () => {
    state = reducer(state, listStyleChangeAct(ListStyle.grid));
    expect(state).toMatchSnapshot();
});
