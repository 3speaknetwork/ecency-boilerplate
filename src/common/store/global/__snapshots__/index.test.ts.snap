// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1- default state 1`] = `
Object {
  "canUseWebp": false,
  "currency": "usd",
  "currencyRate": 1,
  "currencySymbol": "$",
  "filter": "hot",
  "intro": true,
  "listStyle": "row",
  "searchIndexCount": 0,
  "tag": "",
  "theme": "day",
}
`;

exports[`2- location change. should change filter and tag 1`] = `
Object {
  "canUseWebp": false,
  "currency": "usd",
  "currencyRate": 1,
  "currencySymbol": "$",
  "filter": "trending",
  "intro": true,
  "listStyle": "row",
  "searchIndexCount": 0,
  "tag": "bitcoin",
  "theme": "day",
}
`;

exports[`3- theme change 1`] = `
Object {
  "canUseWebp": false,
  "currency": "usd",
  "currencyRate": 1,
  "currencySymbol": "$",
  "filter": "trending",
  "intro": true,
  "listStyle": "row",
  "searchIndexCount": 0,
  "tag": "bitcoin",
  "theme": "night",
}
`;

exports[`4- hide intro 1`] = `
Object {
  "canUseWebp": false,
  "currency": "usd",
  "currencyRate": 1,
  "currencySymbol": "$",
  "filter": "trending",
  "intro": false,
  "listStyle": "row",
  "searchIndexCount": 0,
  "tag": "bitcoin",
  "theme": "night",
}
`;

exports[`5- list style change 1`] = `
Object {
  "canUseWebp": false,
  "currency": "usd",
  "currencyRate": 1,
  "currencySymbol": "$",
  "filter": "trending",
  "intro": false,
  "listStyle": "grid",
  "searchIndexCount": 0,
  "tag": "bitcoin",
  "theme": "night",
}
`;
